src/arrayMethodJoin.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable use-isnan */
+/* eslint-disable no-console */
 'use strict';
 
 /**
  * Implement method join
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const sep = String(separator);
+
+    for (let i = 0; i < this.length; i++) {
+      switch (true) {
+        case (i + 1) !== this.length && this[i] === undefined:
+        case (i + 1) !== this.length && this[i] === null:
+        case (i + 1) !== this.length && this[i] === NaN:
```

Comment:
`NaN` should be just conversed to a string
![image](https://user-images.githubusercontent.com/86923089/226165537-eda3420e-54c0-44cd-ac8c-4f7d2e221bac.png)